python/tank/deploy/tank_commands/setup_project_params.py

```diff
@@ -197,8 +197,13 @@ def set_config_uri(self, config_uri, check_storage_path=True):
         :param check_storage_path: Validate that storage paths exists on disk
         """
         
+        # Toolkit assumes that strings are in fact string objects encoded to UTF-8, not unicode.
+        # In general we avoid sanitizing input within Toolkit, but since this can be wrapped by
+        # custom studio tools and this is a bridge between external input and Toolkit, we make an
+        # exception here and ensure the input is encoded correctly. 
+        config_uri_utf8 = config_uri.encode("utf-8")
```

Comment:
This assumes that the input parameter is a unicode object or a 7 bit ascii str. 

For example, if the input was "Öhrström", you would get:

```
Python 2.7.10 (default, Aug 22 2015, 20:33:39) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> my_name = "Manne Öhrström"
>>> my_name.__class__
<type 'str'>
>>> my_name.encode("utf-8")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 6: ordinal not in range(128)
>>> 

So I think we need a method that can handle all sorts of string like objects, turning them all into utf-8 encoded strings. Maybe something like this - what do you think?

https://github.com/shotgunsoftware/tk-framework-shotgunutils/blob/master/python/shotgun_model/util.py#L101